text/0000-sbom.md

```diff
@@ -0,0 +1,180 @@
+
+## What use cases does it support?
+
+- Consistent SBOM across buildpacks
+- Improved integration with other supply chain analysis and scanning tools
+
+## What is the expected outcome?
+
+- Use of CycloneDX as the SBOM storage format and updates the the specification and lifecycle to reflect the same.
+- Moving the SBOM from `<layer>.toml`, `launch.toml` and `build.toml` to separate files.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Currently the SBOM is defined in the `<layer>.toml`, `launch.toml` and `build.toml` respectively under the `[bom]` table.
+It may not be the most user-friendly way for buildpack authors to create SBOM documents in the above format. This RFC proposes that the `bom` be moved to `<layer>.bom.<ext>`, `launch.bom.<ext>` and `build.bom.<ext>` instead where `<ext>` will be `cdx.json`
+
+The above bom documents need to be created in CycloneDX format.
+
+The lifecycle of the respective `bom` files during rebuilds would be exactly the same as the `bom` table currently. 
+
+The lifecycle would be responsible for taking all of the above `bom` files, merging it while tagging it with the originating layer or buildpack the `bom` came from and storing it in `/layers/config/build.bom.cdx.json` and `/layers/config/launch.bom.cdx.json` respectively for `build` and `launch` boms.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The lifecycle would be responsible for reading, merging and restoring the appropriate `bom` files. The lifecycle of these `bom` files would be tied to their respectice metadata `toml` files. See [RFC 0087](https://github.com/buildpacks/rfcs/blob/main/text/0087-bom-in-layer-metadata.md#how-it-works) for more details. For merging the `bom` files, the lifecycle could replicate or use tooling from [CycloneDX-cli](https://github.com/CycloneDX/cyclonedx-cli) which has a merge operation. The only additional piece of information that the lifecycle would inject are `CycloneDX` [`metadata`](https://cyclonedx.org/use-cases/#properties--name-value-store) the following property keys -
+
+- `io.buildpacks.bom.buildpack.id` - Buildpack ID for the buildpack that created the BOM
+- `io.buildpacks.bom.layer.name` (Optional) - Set to the name of the layer if the `bom` was associated with a specific layer. 
+
+Thw lifecycle will put all the `build` related entries in the `/layers/config/build.bom.cdx.json` file and `launch` entries in `/layers/config/launch.bom.cdx.json`
+
+# Backwards compatibility
+
+For `BOM` from older buildpacks the following mappings will be defined. Each `BOM` entry will be added as a `CycloneDX` component in the final components list.
+
+For eg. if we have the BOM table like below -
+
+```toml
+[[bom]]
+name = "example"
+[bom.metadata]
+version = "0.2.3"
+foo = "bar"
+foo_array = ["bar"]
+[[bom]]
+name = "example2"
+[bom.metadata]
+version = "0.0.1"
+foo = "bar"
+```
+
+It will be converted to - 
```

Comment:
Maybe it's worth adding when it's going to be converted and who is responsible to do so.